packages/block-library/src/media-text/style.scss

```diff
@@ -65,8 +68,8 @@
 	/*!rtl:end:ignore*/
 }
 
-.wp-block-media-text > figure > img,
-.wp-block-media-text > figure > video {
+.wp-block-media-text__media img,
+.wp-block-media-text__media video {
```

Comment:
@jasmussen looks like the styles to handle the image height properly were already present but not applied in the editor because of an extra div. This should fix it